src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Forwarders.cs

```diff
@@ -31,6 +31,7 @@ public ImmutableArray<int> ComplexExtensionMethodInfo
         public bool ContainsForEachStatement => _contextInfo.ContainsForEachStatement;
         public bool ContainsDeconstruction => _contextInfo.ContainsDeconstruction;
         public bool ContainsAwait => _contextInfo.ContainsAwait;
+        public bool ContainsImplicitObjectCreation => _contextInfo.ContainsImplicitObjectCreation;
```

Comment:
i don't think thi si sok.  This is too broad.  We could totally expect 100% of all files to contain this, meaning the search willhave to check all documents.  Can we narrow this down to having the document contain a list of param-counts that it has seen implicit obj creations with?  we can then at least search for docs that are applicable given a constructor with a particular set of params (some of which may be optiona/params)?

Reply:
I'm torn. On one hand, I think this can be done in phases and more optimizations can be added later. On the other hand, I'm not sure that partitioning on argument count would address your concern (most constructors probably have 1-4 parameters), so the perf may not be improved that much by that design.
Would it be safer if I remove FAR handling for now?

Reply:
Yes. i'm actually pretty concerned about this on the whole.  We may jsut need an entirely different system for "find me implement references" (which can be very expensive, but which users explicitly ask for).

Reply:
Okay. I'm reverting the FAR changes for now.